status/controller.go

```diff
@@ -23,24 +25,52 @@ import (
 )
 
 type Controller[T Object] struct {
-	kubeClient         client.Client
-	eventRecorder      record.EventRecorder
-	observedConditions sync.Map // map[reconcile.Request]ConditionSet
-	terminatingObjects sync.Map // map[reconcile.Request]DeletionTimestamp
+	objectGVK                     schema.GroupVersionKind
+	kubeClient                    client.Client
+	eventRecorder                 record.EventRecorder
+	observedConditions            sync.Map // map[reconcile.Request]ConditionSet
+	terminatingObjects            sync.Map // map[reconcile.Request]DeletionTimestamp
+	ConditionDuration             pmetrics.ObservationMetric
+	ConditionCount                pmetrics.GaugeMetric
+	ConditionCurrentStatusSeconds pmetrics.GaugeMetric
+	ConditionTransitionsTotal     pmetrics.CounterMetric
+	TerminationCurrentTimeSeconds pmetrics.GaugeMetric
+	TerminationDuration           pmetrics.ObservationMetric
 }
 
-func NewController[T Object](client client.Client, eventRecorder record.EventRecorder) *Controller[T] {
+type ControllerOpts struct {
+	GVK schema.GroupVersionKind
+}
+
+func NewController[T Object](client client.Client, eventRecorder record.EventRecorder, opts ...ControllerOpts) *Controller[T] {
+	var gvk schema.GroupVersionKind
+	switch len(opts) {
+	case 0:
+		gvk = object.GVK(object.New[T]())
+	case 1:
+		gvk = opts[0].GVK
+	default:
+		panic("expected no or one argument for controller options")
```

Comment:
In my mind, I could not think of a case where we would need to pass in more then one controller option